sdks/go/pkg/beam/core/graph/fn.go

```diff
@@ -816,11 +849,17 @@ func validateSdfSigNumbers(fn *Fn, num int) error {
 		splitRestrictionName:         num + 1,
 		restrictionSizeName:          num + 1,
 		createTrackerName:            1,
+		truncateRestrictionName:      num + 1,
 	}
 	returnNum := 1 // TODO(BEAM-3301): Enable optional error params in SDF methods.
 
-	for _, name := range requiredSdfNames {
-		method := fn.methods[name]
+	optionalSdfs := optionalSdfNameMap()
+	for _, name := range sdfNames {
+		method, ok := fn.methods[name]
+		if !ok && optionalSdfs[name] {
+			// skip validating unimplemented optional sdf methods
+			continue
+		}
```

Comment:
```suggestion
	for _, name := range requiredSdfNames {
		method, ok := fn.methods[name]
```

Rather than using the optional map, we can just use the list of requiredSdfs that we already have